### PR TITLE
Upsert NotionDatabases in connectorsDB with title and url

### PR DESCRIPTION
**Pull request by spolu**
This PR adds `lastSeenTs`, `parentType`, `perentId`, `title`, `notionUrl` to the `NotionDatabase` connectors model.

The `lastSeenTs` is used to properly garbage collect Notion Databases
The other will be used to properly show Databases in the Dust Notion permissions

A few comments and bug fixes were made along the way (moving to object signature at places to avoid mixing up arguments)

### CODE REVIEW

**Comment by spolu** [comment]
Deploy involves:
- deploying connectors
- running the `notion restart-all` command from the connectors cli (so that the new workflow use the new queues)

File: connectors/src/connectors/notion/lib/connectors_db_helpers.ts [hunk 1]
```diff
@@ -1,17 +1,27 @@
...
-  notionPageId: string,
-  lastSeenTs: number,
-  parentType?: string | null,
-  parentId?: string | null,
-  title?: string | null,
-  notionUrl?: string | null,
-  lastUpsertedTs?: number,
-  skipReason?: string
-): Promise<NotionPage> {
+export async function upsertNotionPageInConnectorsDb({
```
Comment (fontanierh) [inline connectors/src/connectors/notion/lib/connectors_db_helpers.ts:4]: 🙏

File: connectors/src/connectors/notion/temporal/activities.ts [hunk 2]
```diff
@@ -452,23 +579,51 @@ export async function garbageCollectActivity(
...
-    "Found pages to delete."
+    {
+      pagesToDeleteCount: pagesToDelete.length,
+      databasesToDeleteCount: databasesToDelete.length,
+    },
+    "Found pages and databases to delete."
   );
 
-  const notionAccessToken = await getNotionAccessToken(connector.connectionId);
+  const NOTION_UNHEALTHY_ERROR_CODES = [
```
Comment (fontanierh) [inline connectors/src/connectors/notion/temporal/activities.ts:619]: 👍

**Review by fontanierh — DISMISSED** [review]
Looks great ! Thanks for all the improvements to the notion code too 🙏 🙏

**Comment by spolu** [comment]
Cool will test thoroughly. @fontanierh May I ask that you do a second-pass review really squinting for potential problems? :pray:

**Comment by spolu** [comment]
Things I still wanted to check:
- make sure we do ignore only notion database with a skip reason to make sure creating record won't ignore everything
- rethink about GC before full-resync make sure everything looks sane

Will test:
- deploying and see if running as is incremental updates is OK
- (ideally) running a GC but not sure I'll go that far (worst case we bad GC databases I believe)
- running a full resync

**Comment by spolu** [comment]
- incremental sync post merge tested
- GC post merge tested
- full-resync tested

**Comment by spolu** [comment]
I will go ahead and merge this and restart our workspace

**Comment by spolu** [comment]
Deploy is:
- merge
- init connector db
- deploy connectors

**Pull request merged**